### PR TITLE
release: sourcegraph@4.0.0

### PR DESCRIPTION
**Pull request by camdencheek**
This pull request is part of the Sourcegraph 4.0.0 release.

* [Release batch change](https://k8s.sgdev.org/organizations/sourcegraph/batch-changes/release-sourcegraph-4.0.0)
* [Tracking issue](https://github.com/sourcegraph/sourcegraph/issues/41759)

### Test plan

CI checks in this repository should pass, and a manual review should confirm if the generated changes are correct.

### :warning: Additional changes required

These steps must be completed before this PR can be merged, unless otherwise stated. Push any required changes directly to this PR branch.

- [x] Ensure all other pull requests in the batch change have been merged
- [x] Run `yarn run release release:finalize` to generate the tags required. CI will not pass until this command is run.
- [x] Re-run the build on this branch (using either `sg ci build --wait` or the Buildkite UI) and merge when the build passes.

cc @camdencheek

### CODE REVIEW

File: doc/dev/background-information/ci/reference.md [hunk 1]
```diff
@@ -90,8 +90,8 @@ The run type for environment including `{"RELEASE_NIGHTLY":"true"}`.
 Base pipeline (more steps might be included based on branch changes):
 
 - **Metadata**: Pipeline metadata
-- Trigger 3.43 release branch healthcheck build
-- Trigger 3.42 release branch healthcheck build
+- Trigger 4.0 release branch healthcheck build
+- Trigger 4.-1 release branch healthcheck build
```
Comment (coury-clark) [inline doc/dev/background-information/ci/reference.md:94]: Whats going on here?

Comment (camdencheek) [reply doc/dev/background-information/ci/reference.md:94]: That is generated by `sg generate`. It subtracts 1 from the minor version, giving -1 in this case. I should double check that we won't actually try to do a `4.-1` release branch healthcheck build though

Comment (camdencheek) [reply doc/dev/background-information/ci/reference.md:94]: Actually, it looks like this is not just documentation, but will actually attempt to run an invalid test.

Comment (camdencheek) [reply doc/dev/background-information/ci/reference.md]: Fixed in 3f70c6a. cc @sourcegraph/dev-experience, this is kinda hacky, but I'd like to unblock the release on this. Does this look like an okay solution for right now?

**Review by bobheadxi — APPROVED** [review]

**Review by coury-clark — APPROVED** [review]

**Pull request merged**